src/Eventso.Subscription.Hosting/NoDeadLetterQueue.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+using Eventso.Subscription.Kafka.DeadLetter.Store;
+using Eventso.Subscription.Observing.DeadLetter;
+
+namespace Eventso.Subscription.Hosting
+{
+    internal sealed class NoDeadLetterQueue : IDeadLetterQueue, IDeadLetterQueueScopeFactory, IPoisonEventStore
+    {
+        public static readonly string ErrorMessage =
+            $"Dead letter queue was not configured, please refer to {nameof(ServiceCollectionExtensions.AddSubscriptions)} method.";
+
+        public static readonly NoDeadLetterQueue Instance = new();
+        
+        public void Enqueue(DeadLetter message)
+            => throw new InvalidOperationException(ErrorMessage);
```

Comment:
so if someone makes mistake in the dlq configuration, he will find out that something is wrong only when the first poison message arrives? it could happen in night time :)
I think NoDeadLetterQueue is pointless, we should throw an exception at startup in case of any inconsistencies in dlq configuration